### PR TITLE
Chore: add integration tests for autofixing (fixes #5909)

### LINKED ISSUE #5909: Chore: Create integration test for auto-fixing
JSCS had a very nice test to check auto-fixing. It had two config files and it would take the whole source code of itself and autofix it to second config, then autofix it to first and compare results. @mikesherov @hzoo @markelog @mdevils @zxqfox correct me if I'm wrong in my description above.

I think it would be really nice to add similar test to ESLint. We just added multi-pass autofixing, and it would be nice to verify that it's working correctly.

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [x] I've included tests for my change
- (n/a) I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

This adds integration tests for autofixing files, as described in https://github.com/eslint/eslint/issues/5909.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular.

### CODE REVIEW

File: tests/bin/eslint.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+const fs = require("fs");
 const assert = require("chai").assert;
 const EXECUTABLE_PATH = require("path").resolve(`${__dirname}/../../bin/eslint.js`);
 
 /**
+* Returns a Promise for when a child process exits
+* @param {ChildProcess} exitingProcess The child process
+* @returns {Promise} A Promise that fulfills when the child process exits
+*/
+function awaitExit(exitingProcess) {
```
Comment (btmills) [inline tests/bin/eslint.js]: `assertExitCode` uses this same logic as part of its implementation. Is it worth changing it to call `awaitExit`? `awaitExit` would have to be modified to resolve with the exit code.

```js
function awaitExit(exitingProcess) {
    return new Promise(resolve => exitingProcess.once("exit", resolve));
}
// ...
function assertExitCode(exitingProcess, expectedExitCode) {
    return awaitExit(exitingProcess).then(exitCode => {
        if (exitCode !== expectedExitCode) {
            throw new Error(`Expected an exit code of ${expectedExitCode} but got ${exitCode}.`);
        }
    });
}
```

Comment (not-an-aardvark) [inline tests/bin/eslint.js]: Good idea. I'll change that and update this PR.

**Comment by kaicataldo** [comment]
I read the original issue differently:
> It had two config files and it would take the **whole source code of itself** and autofix it to second config

Sounds like the integration test ran on the actual source code? Maybe I'm reading that wrong?

**Comment by vitorbal** [comment]
I actually interpreted it like that as well. However I'm not opposed to
this version to start off with, especially if the original idea would be
way much work.

**Comment by kaicataldo** [comment]
@vitorbal Agreed :+1:

**Comment by platinumazure** [comment]
We can definitely improve this later, so I have no issue with this going in as is (this is not a review, lol).

Another aspect that might have been missed here, but which we can consider later: Auto-fix a file from Config A to Config B, and then auto-fix again *back* from Config B to Config A, to make sure we get the same initial result. This would help us detect potential asymmetry in the rule fixers. But again, I'm in favor of incremental improvements here.

**Pull request merged**